### PR TITLE
deprecate no_cuda

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?
This PR deprecates the `no_cuda` arg because it is confusing for Mac users as their models get dispatched to `mps` device when `no_cuda=False`. If they want to train the model on cpu, they need to set `no_cuda=True` which is not intuitive. We rename it `use_cpu` instead. 

Related issue #24697

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -298,7 +298,9 @@ class TrainingArguments:
             This should not be activated when the different nodes use the same storage as the files will be saved with
             the same names for each node.
         no_cuda (`bool`, *optional*, defaults to `False`):
-            Whether to not use CUDA even when it is available or not.
+            This argument is deprecated. Use `use_cpu` instead. Do not use CUDA even when it is available
```
Comment (sgugger) [inline src/transformers/training_args.py:301]: For deprecated arguments, we remove the doc entirely.

**Review by sgugger — APPROVED** [review]
Thanks for adding this! Just one comment on the docstring.

Please hold off merging until the release branch has been cut.

File: src/transformers/training_args.py [hunk 2]
```diff
@@ -298,7 +298,9 @@ class TrainingArguments:
             This should not be activated when the different nodes use the same storage as the files will be saved with
             the same names for each node.
         no_cuda (`bool`, *optional*, defaults to `False`):
-            Whether to not use CUDA even when it is available or not.
+            This argument is deprecated. Use `use_cpu` instead.
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply src/transformers/training_args.py:301]: Oh my bad ;)

**Review by sgugger — APPROVED** [review]
Perfect!

**Pull request merged**